entity-framework/core/what-is-new/ef-core-7.0/whatsnew.md

```diff
@@ -1184,3 +1129,532 @@ dotnet new ef-templates
+        IConventionEntityTypeBuilder entityTypeBuilder,
+        IConventionContext<IConventionEntityTypeBuilder> context)
+        => Process(entityTypeBuilder);
+
+    public override void ProcessEntityTypeBaseTypeChanged(
+        IConventionEntityTypeBuilder entityTypeBuilder,
+        IConventionEntityType? newBaseType,
+        IConventionEntityType? oldBaseType,
+        IConventionContext<IConventionEntityType> context)
+    {
+        if ((newBaseType == null
+             || oldBaseType != null)
+            && entityTypeBuilder.Metadata.BaseType == newBaseType)
+        {
+            Process(entityTypeBuilder);
+        }
+    }
+
+    private void Process(IConventionEntityTypeBuilder entityTypeBuilder)
+    {
+        foreach (var memberInfo in GetRuntimeMembers())
+        {
+            if (Attribute.IsDefined(memberInfo, typeof(PersistAttribute), inherit: true))
+            {
+                entityTypeBuilder.Property(memberInfo);
+            }
+            else if (memberInfo is PropertyInfo propertyInfo
+                     && Dependencies.TypeMappingSource.FindMapping(propertyInfo) != null)
+            {
+                entityTypeBuilder.Ignore(propertyInfo.Name);
+            }
+        }
+
+        IEnumerable<MemberInfo> GetRuntimeMembers()
+        {
+            var clrType = entityTypeBuilder.Metadata.ClrType;
+
+            foreach (var property in clrType.GetRuntimeProperties()
+                         .Where(p => p.GetMethod != null && !p.GetMethod.IsStatic))
+            {
+                yield return property;
+            }
+
+            foreach (var property in clrType.GetRuntimeFields())
+            {
+                yield return property;
+            }
+        }
+    }
+}
+-->
+[!code-csharp[AttributeBasedPropertyDiscoveryConvention](../../../../samples/core/Miscellaneous/NewInEFCore7/ModelBuildingConventionsSample.cs?name=AttributeBasedPropertyDiscoveryConvention)]
+
+> [!TIP]
+> When replacing a built-in convention, the new convention implementation should inherit from the existing convention class.
+
+The convention is then registered using the `Replace` method in `ConfigureConventions`:
+
+<!--
+    protected override void ConfigureConventions(ModelConfigurationBuilder configurationBuilder)
+    {
+        configurationBuilder.Conventions.Replace<PropertyDiscoveryConvention>(
+            serviceProvider => new AttributeBasedPropertyDiscoveryConvention(
+                serviceProvider.GetRequiredService<ProviderConventionSetBuilderDependencies>()));
+    }
+-->
+[!code-csharp[ReplaceConvention](../../../../samples/core/Miscellaneous/NewInEFCore7/ModelBuildingConventionsSample.cs?name=ReplaceConvention)]
+
+> [!TIP]
+> This is a case where the existing convention has dependencies, represented by the `ProviderConventionSetBuilderDependencies` dependency object. These are obtained from the internal service provider using `GetRequiredService` and passed to the convention constructor.
```

Comment:
Probably should also mention that if there is a provider-specific version of a convention it is base to replace that type, than the base as it might implement additional convention interfaces